### ISSUE TITLE
"Calling the Classcraft API" returns a response code, not a JSON

### ISSUE THREAD

**Issue by thorbengrubert** [classcraft/classcraft-api-issues#3]
When following the step "Calling the Classcraft API" of the documentation (https://developers.classcraft.com/documentation/oauth), the response is a code instead of a JSON file.

![code_response](https://user-images.githubusercontent.com/45844445/90233829-035cfe80-de1f-11ea-8709-74991e2bb8e9.jpg)

Here are the specific requests used:
For the auth code: https://auth.classcraft.com?client_id=rN2cgLhuzD4yRCkLM&redirect_uri=http://localhost:8080&scope=students&response_type=code
For the token: https://auth.classcraft.com/token?code=200f3a76fcd8dd0575e0ddaea57b42ab08dce629&client_id=rN2cgLhuzD4yRCkLM&client_secret=E00VU0Zw5KHMWU5CmlkF9oR9&redirect_uri=http://localhost:8080&grant_type=code&scope=students&response_type=code

**Comment by simonbelanger** [comment]
Hi! The redirection URL displayed in your screenshot contains a `code` parameter. This redirection requests must be handled by your backend to exchange this auth code for tokens by calling the `https://auth.classcraft.com/token?[...]` endpoint.

**Comment by thorbengrubert** [comment]
Hi Simon,
thanks for the quick reply!

Okay, so far I think I've understood what the steps are. I am using the code I have received to then call auth.classcraft.com/token (see the last link in my previous post). What puzzles me here is that a response type is seemingly required although it is not part of the documentation and, if left out, results in an error: "Missing parameter: `response_type`."

It seems I am missing something here, am I not?

**Comment by simonbelanger** [comment]
that's weird 🤔 ... `response_type` is a required parameter for `https://auth.classcraft.com`, not `https://auth.classcraft.com/token`... are you sure you are calling the right URL?

**Comment by thorbengrubert** [comment]
Hi Simon,
sorry for the delay in my response. Here's where I'm at.

I start with the initial request: https://auth.classcraft.com?client_id=rN2cgLhuzD4yRCkLM&redirect_uri=http://localhost:8080&scope=me.readonly&response_type=code

And get a response: http://localhost:8080/?code=852d839223748a540bc8c0f1db375c975d4c4d29

Then, for the token: https://auth.classcraft.com/token?code=073cbfc5950a7608f40b251f53fc4f1484dbbff5&client_id=rN2cgLhuzD4yRCkLM&client_secret=oMgYH8w6zTHqRJp4USL21V33&redirect_uri=http://localhost:8080&grant_type=authorization_code

But my response is an error: https://auth.classcraft.com/error
![ScopeParameterMissing](https://user-images.githubusercontent.com/45844445/91399745-d0633380-e83e-11ea-91c7-a53bd361e26c.png)

If I I then also provide the scope parameter (e.g. "&scope=students"), we are back to where we left off in my previous post: "response_type missing".

I went through the API documentation step-by-step a couple of times but I struggle to pin down where I must have deviated from the instructions.

**Comment by simonbelanger** [comment]
The value passed to the `scope` parameter should be the same for both the initial request and the token one. My understanding is that you're specifying `me.readonly` for the initial request and `students` for the token one. Can you try using the same value for both request?

Although I must admit that the "response_type missing" error is weird 🤔 .

**Comment by thorbengrubert** [comment]
Good idea, in my reponse I missed that I've just two different scope parameters!
I've just tried using "scope=me.readonly" in both requests, however, I'm afraid it doesn't change the fact that I still got the "response_type missing" error.

**Comment by simonbelanger** [comment]
Out of curiosity, have you tried to include `response_type=code` in the token request?

**Comment by thorbengrubert** [comment]
Interestingly, I then get another code: http://localhost:8080/?code=efc0ec08477b9e0ded318459871c98305aaca4a9

**Comment by simonbelanger** [comment]
Are you sending the token request as POST and not GET?

**Comment by thorbengrubert** [comment]
... yes, accidentally the entire time as POST - and this might very well fix it. I'll try out a few things and report back but I'm quite hopeful that this was the culprit.
